data/unmarshal.go

```diff
@@ -0,0 +1,304 @@
+package data
+
+import (
+	"errors"
+	"fmt"
+	"math"
+
+	"github.com/ipld/go-ipld-prime"
+	"github.com/ipld/go-ipld-prime/fluent/qp"
+	"google.golang.org/protobuf/encoding/protowire"
+)
+
+func DecodeUnixFSData(src []byte) (UnixFSData, error) {
```

Comment:
Is there a reason not to follow the `Decoder func(NodeAssembler, io.Reader) error` interface here?

Reply:
yea kinda -- it's never going to be registered as a multicodec, and this provides a bunch of weird type casts and artificial reader constructions all around. This isn't an IPLD format per se. We will always be decoding from within a larger block.